code/src/nuvla/connector/kubernetes.py

```diff
@@ -535,6 +539,12 @@ def __init__(self, job: Job, **kwargs):
         self.job = job
         self.api = job.api
 
+        self.nuvlabox_id = kwargs.get("nuvlabox_id")
+        self.nuvlabox_resource = self.api.get(self.nuvlabox_id)
+        self.nuvlabox = self.nuvlabox_resource.data
+        self.nuvlabox_status = self.api.get( \
+            self.nuvlabox.get("nuvlabox-status")).data
```

Comment:
What if there is no `nuvlabox_id` and so `self.nuvlabox_id` is `None` ?
Same with `nuvlabox-status`.\
I think some conditions should be added.

Reply:
How about applying the test way back at https://github.com/nuvla/job-engine/blob/d14dc6d21ea85669b28d0bdf03be22da52d606e8/code/src/nuvla/job/actions/nuvlabox_update.py#L17

as the check is valid for both kubernetes and docker installations?

Reply:
I think it depends how far we want the `job` object to propagate.
I prefer that the `job` do not go further than the "`action`" so that it easier to use the other modules (like `nuvlabox.py` and `kubernetes.py`) without a job context, for example for a unit test.